### ISSUE TITLE
App freezes when adding a new Auto upload folder

### ISSUE THREAD

**Issue by devvykid** [nextcloud/android#6310]
### Steps to reproduce
1. Go to Sidebar > Auto uploads
2. Click on any folder's Cloud icon or turn the upload switch on
3. 

### Expected behaviour
- It should not freeze and sync normally.

### Actual behaviour
- When I restart the app, the cloud icon is turned blue, but it doesn't sync.
- If the folder has few images, the app doesn't freeze completely. Instead, it freezes for a couple of seconds and I was able to contol the app again.

### Can you reproduce this problem on https://try.nextcloud.com?
- Please create a test demo account and see if this still happens there.
- If yes, please open up a bug report
- If not, please verify server setup and ask for help on forum

Yes.

### Environment data
Android version: 9.0 Pie

Device model: SM-N935K (Samsung Galaxy Note FE)

Stock or customized system: Stock

Nextcloud app version: latest (3.12.0)

Nextcloud server version: 18.0.4

Reverse proxy: php7.4-fpm (deb.sury.org)

#### Web server error log
```
Insert your webserver log here
```

#### Nextcloud log (data/nextcloud.log)
```
Insert your Nextcloud log here
```
**NOTE:** Be super sure to remove sensitive data like passwords, note that everybody can look here! You can use the Issue Template application to prefill some of the required information: https://apps.nextcloud.com/apps/issuetemplate

**Comment by devvykid** [comment]
Seems similar to #6287.

**Comment by tobiasKaminsky** [comment]
This is fixed in master, and will be in 3.13.
But I think it is worth to backport it to upcoming 3.12.1.

**Comment by joshtrichards** [comment]
Fixed in #6328
